Fixes AttributeError in terrain-based position command

abmoRobotics opened the pull request:
# Description
Fixes AttributeError in position_command.py
Fixes #189 

<!-- As a practice, it is recommended to open an issue to have discussions on the proposed pull request.
This makes it easier for the community to keep track of what is being developed or added, and if a given feature
is demanded by more than one party. -->

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## Screenshots

Please attach before and after screenshots of the change if applicable.

<!--
Example:

| Before | After |
| ------ | ----- |
| _gif/png before_ | _gif/png after_ |

To upload images to a PR -- simply drag and drop an image while in edit mode and it should upload the image directly. You can then paste that source into the above before/after sections.
-->

## Checklist

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `./orbit.sh --format`
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the changelog and the corresponding version in the extension's `config/extension.toml` file
- [x] I have added my name to the `CONTRIBUTORS.md` or my name already exists there

<!--
As you go through the checklist above, you can mark something as done by putting an x character in it

For example,
- [x] I have done this task
- [ ] I have not done this task
-->

Mayankm96 commented:
Thanks for sending the MR. While it makes sense, currently, I don't think the terrain-based posisiton command term even works. We are working on fixing this up and having an example for it. This issue should automatically be resolved with it.

Mayankm96 commented:
Yes, that should be the correct way to use the class :)

We should probably make a dummy function in the current TerrainImporter that returns the origins as the commands. At least that way, the command term still works.

If you can fix that, we can merge this to at least have it in a working state. Also please update the `extension.toml` and `CHANGELOG.md` file :)

abmoRobotics commented:
Perfect :) 
I have added a dummy function ```sample_new_targets(self, env_ids)``` that returns the env_origins as commands to ```TerrainImporter```, and updated ```extension.toml``` aswell as ```CHANGELOG.md```. The command term now works without extending the ```TerrainImporter``` - let me know if there is anything else that needs to be done before merging.

Mayankm96 commented on source/extensions/omni.isaac.orbit/docs/CHANGELOG.rst:
```diff
@@ -1,5 +1,13 @@
+Fixed
+^^^^^
+
+* Fixed AttributeError in :class:`omni.isaac.orbit.env.TerrainBasedPositionCommand.` and added dummy function that returns
+  env_origins as terrain based targets in :class:`omni.isaac.orbit.terrain.TerrainImporter`.
```
```suggestion
* Fixed attribute error raised when calling the :class:`omni.isaac.orbit.envs.mdp.TerrainBasedPositionCommand` command term.
* Added a dummy function in :class:`omni.isaac.orbit.terrain.TerrainImporter` that returns environment origins as terrain-aware sampled targets. This function should be implemented by child classes based on the terrain type.
```

Mayankm96 commented on source/extensions/omni.isaac.orbit/omni/isaac/orbit/terrains/terrain_importer.py:
```diff
@@ -343,3 +343,11 @@ def _compute_env_origins_grid(self, num_envs: int, env_spacing: float) -> torch.
         env_origins[:, 2] = 0.0
         return env_origins
+
+    """
+    Dummy Functions.
+    """
```
Can you move this to the section above "Internal helpers"? We try to keep private methods in the end and operations before.

You can call it "Operations - Sampling" instead of "Dummy".

abmoRobotics replied:
I have moved the section above the "Internal helpers", and renamed it in [e05bf0](https://github.com/NVIDIA-Omniverse/orbit/pull/191/commits/e075bf0ecbb4655f38ad46d9d161ab729e09c558).

Mayankm96 commented on source/extensions/omni.isaac.orbit/omni/isaac/orbit/terrains/terrain_importer.py:
```diff
@@ -343,3 +343,11 @@ def _compute_env_origins_grid(self, num_envs: int, env_spacing: float) -> torch.
+    Dummy Functions.
+    """
+
+    def sample_new_targets(self, env_ids: torch.Tensor) -> torch.Tensor:
+        """Returns the environment origins as target locations."""
```
```suggestion
    def sample_new_targets(self, env_ids: Sequence[int]) -> torch.Tensor:
        """Samples terrain-aware locations of flat patches to set spawn or target locations.
        
        Note:
             This is a dummy implementation that returns the environment origins as target locations. Please inherit the class and reimplement the function for specific terrain types.
         
         Args:
             env_ids: The environment indices to sample target locations for.
          
          Returns:
              The sampled target locations as (x, y, z, yaw). Shape is (N, 4).
         """
```

abmoRobotics replied:
Updated the docstring in [e075bf0](https://github.com/NVIDIA-Omniverse/orbit/pull/191/commits/e075bf0ecbb4655f38ad46d9d161ab729e09c558), it returns (x, y, z) of shape (N, 3) right?

Mayankm96 commented on source/extensions/omni.isaac.orbit/docs/CHANGELOG.rst:
```diff
@@ -1,5 +1,13 @@
+Fixed
+^^^^^
+
+* Fixed attribute error raised when calling the :class:`omni.isaac.orbit.envs.mdp.TerrainBasedPositionCommand` command term.
+* Added a dummy function in :class:`omni.isaac.orbit.terrain.TerrainImporter` that returns environment origins as terrain-aware sampled targets. This function should be implemented by child classes based on the terrain type.
```
```suggestion
* Fixed attribute error raised when calling the :class:`omni.isaac.orbit.envs.mdp.TerrainBasedPositionCommand` 
   command term.
* Added a dummy function in :class:`omni.isaac.orbit.terrain.TerrainImporter` that returns environment
   origins as terrain-aware sampled targets. This function should be implemented by child classes based on
   the terrain type.
```

Mayankm96 commented on source/extensions/omni.isaac.orbit/omni/isaac/orbit/terrains/terrain_importer.py:
```diff
@@ -304,6 +304,25 @@ def update_env_origins(self, env_ids: torch.Tensor, move_up: torch.Tensor, move_
+    def sample_new_targets(self, env_ids: Sequence[int]) -> torch.Tensor:
+        """Samples terrain-aware locations of flat patches to set spawn or target locations.
+
+        Note: This is a dummy function that returns the environment origins as target locations.
+        Please inherit the class and reimplement the function for specific terrain types
```
```suggestion
        Note:
            This is a dummy function that returns the environment origins as target locations.
            Please inherit the class and reimplement the function for specific terrain types
```

Mayankm96 commented on source/extensions/omni.isaac.orbit/docs/CHANGELOG.rst:
```diff
@@ -1,5 +1,16 @@
 Changelog
 ---------
+0.10.11 (2024-01-08)
+~~~~~~~~~~~~~~~~~~~~
+
+Fixed
+^^^^^
+
+* Fixed attribute error raised when calling the :class:`omni.isaac.orbit.envs.mdp.TerrainBasedPositionCommand` 
+   command term.
+* Added a dummy function in :class:`omni.isaac.orbit.terrain.TerrainImporter` that returns environment
+   origins as terrain-aware sampled targets. This function should be implemented by child classes based on
+   the terrain type.
 
```
```suggestion
0.10.11 (2024-01-08)
~~~~~~~~~~~~~~~~~~~~

Fixed
^^^^^

* Fixed attribute error raised when calling the :class:`omni.isaac.orbit.envs.mdp.TerrainBasedPositionCommand` 
  command term.
* Added a dummy function in :class:`omni.isaac.orbit.terrain.TerrainImporter` that returns environment
  origins as terrain-aware sampled targets. This function should be implemented by child classes based on
  the terrain type.


```

Mayankm96 commented on source/extensions/omni.isaac.orbit/docs/CHANGELOG.rst:
```diff
@@ -1,5 +1,17 @@
+Fixed
+^^^^^
+
+* Fixed attribute error raised when calling the :class:`omni.isaac.orbit.envs.mdp.TerrainBasedPositionCommand` 
+  command term.
```
```suggestion
* Fixed attribute error raised when calling the :class:`omni.isaac.orbit.envs.mdp.TerrainBasedPositionCommand`
  command term.
```

Mayankm96 reviewed (approved):
Looks good to me!
